corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -287,11 +287,10 @@ def _add_menu_instances(self, menu):
             xpaths.add(menu.relevant)
 
         known_instances, unknown_instance_ids = get_all_instances_referenced_in_xpaths(self.app, xpaths)
+        assert_no_unknown_instances(unknown_instance_ids)
```

Comment:
This gives me pause, and I'd appreciate a second opinion.  This code only looks for "standard" instances, but prior to my changes here, menu relevancy conditions could reference custom instances.  These were defined on forms and pulled from module search config properties.  Those will still be added to forms regardless of whether they're referenced anywhere (see below comment), but it feels more than a bit scattered, and I'm worried there might be gaps for things to slip through
https://github.com/dimagi/commcare-hq/blob/387a2f423acbb833fbe2346b563ac6c1fded1f17/corehq/apps/app_manager/suite_xml/post_process/instances.py#L214

Reply:
What is the difference between custom and standard instances here ? And your change remove custom instances from `menu` blocks and not from `entry` right ? 

I think we do want to follow the same Xpath evaluation pattern for both relevancy and assertion Xpaths but to be backward compatible FP will now need to look at both `entry` and `menu` instance connectors to form the full list of instances. So as long as these instances are present in one of `entry` or `menu` it should be good.

Reply:
> What is the difference between custom and standard instances here ?

https://github.com/dimagi/commcare-hq/blob/d0c8a0f0f79444921190f39166567a2762ecf502/corehq/apps/app_manager/suite_xml/post_process/instances.py#L53-L58

"Standard" instances are those defined here and in that same file via the `@register_factory` decorator
https://github.com/dimagi/commcare-hq/blob/d0c8a0f0f79444921190f39166567a2762ecf502/corehq/apps/app_manager/suite_xml/post_process/instances.py#L310-L317

> And your change remove custom instances from menu blocks and not from entry right ?

correct

> So as long as these instances are present in one of entry or menu it should be good.

I guess that means any current menu relevancy conditions using a custom instance declaration are definitely doing a weird workaround to make it work anyways (using a simprints flag and adding declarations to a form in the menu or adding a noop case search lookup table question), and assuming the mobile code continues to load entry instances, that'll still work.